src/d3a/models/market/one_sided.py

```diff
@@ -82,7 +84,7 @@ def offer(self, price: float, energy: float, seller: str, seller_origin,
         offer = Offer(offer_id, price, energy, seller, original_offer_price,
                       seller_origin=seller_origin)
 
-        self.offers[offer.id] = deepcopy(offer)
+        self.offers[offer.id] = offer
```

Comment:
Hmm, but arent we falling into the memory leak? I dont remember, but are we deleting the connection from market to the offers before deleting the markets too?

Reply:
This was mostly a precaution, especially after removing the markets/agents offers when recycling the market. I did not observe any increased memory when removing this, but I'll give it one more check to be on the safe side. Thanks

Reply:
ahh, wow. Ok, I hope that it is indeed only a precaution and we can omit the deepcopy. Thanks for checking again!